crates/bevy_ecs/Cargo.toml

```diff
@@ -85,6 +85,8 @@ critical-section = [
 
 hotpatching = ["dep:subsecond"]
 
+debug = []
```

Comment:
To keep complexity and boilerplate down, maybe we could do something like:

```rust
#[cfg(feature = "debug")]
struct DebugValue<T>(T);

#[cfg(not(feature = "debug")]
struct DebugValue<T>(PhantomData<T>);

struct Something {
  some_debug_thing: DebugValue<String>(),
}

let x = Something {
  some_debug_string: debug_value!(std::any::type_name::<Something>().to_string()),
}
```

Where `debug_value!` becomes just `DebugValue<String>(PhantomData)` when debug is disabled and `DebugValue(std::any::type_name::<Something>().to_string())` when it is enabled.

Likewise, this has overlap with the new `bevy_platform::cfg` stuff. Perhaps we could use a `bevy_platform::cfg::debug!()`?